ui/components/report.jsx

```diff
@@ -1,8 +1,28 @@
-var React = require('react');
+var React = require('react'),
+    Header = require('./header.jsx');
 
 var Report = React.createClass({
   render: function() {
-    return null;
+    var data = this.props.data,
+        passes = 0,
+        failures = 0,
+        duration = 0;
+
+    for (var i = 0; i < data.length; i++) {
+      for (var j = 0; j < data[i].tests.length; j++) {
+        if (data[i].tests[j].state === 'passed') {
```

Comment:
Use [filter](http://devdocs.io/javascript/global_objects/array/filter). Functional programming is all the rage.